packages/cli/src/commands/types/generate/actions.ts

```diff
@@ -296,133 +296,205 @@ async function loadCompilerOptions(path: string) {
   return {};
 }
 
-export const generateTypes = async (
+const getRequiredFields = (schema: Record<string, any> | undefined): string[] => {
+  return Object.entries(schema || {}).reduce<string[]>(
+    (acc, [_, v]) => (v && typeof v === 'object' && 'required' in v && (v as any).required ? [...acc, _] : acc),
+    ['component', '_uid'],
+  );
+};
+
+const collectUsedTypesFromProps = (
+  props: JSONSchema['properties'] | undefined,
+  used: Set<string>,
+) => {
+  if (!props) return;
```

Comment:
question: why is this check here and not where this method is called?

Reply:
It was to resolve this